casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala

```diff
@@ -152,36 +148,31 @@ object InterpreterUtil {
     } else {
       //In the case of multiple parents we need
       //to apply all of the deploys that have been
-      //made in all histories since the greatest
-      //common ancestor in order to reach the current
-      //state.
-      for {
-        gca <- parents.toList.foldM(parents.head) {
-                case (gca, parent) =>
-                  DagOperations.greatestCommonAncestorF[F](gca, parent, genesis, dag)
-              }
-
-        gcaStateHash = ProtoUtil.tuplespace(gca).get
-        _ = assert(
-          knownStateHashes.contains(gcaStateHash),
-          "We should have already computed state hash for GCA when we added the GCA to our blockDAG.")
-
-        // TODO: Have proper merge of tuplespaces instead of recomputing.
-        ancestors <- DagOperations
-                      .bfTraverseF[F, BlockMessage](parentTuplespaces.map(_._1).toList) { block =>
-                        if (block == gca) List.empty[BlockMessage].pure[F]
-                        else ProtoUtil.unsafeGetParents[F](block)
-                      }
-                      .filter(_ != gca) //do not include gca deploys
-                      .toList
-
-        deploys = ancestors
-          .flatMap(ProtoUtil.deploys(_).reverse.flatMap(ProcessedDeployUtil.toInternal))
-          .toIndexedSeq
-          .reverse
+      //made in all of the branches of the DAG being
```

Comment:
This function is pretty big and, as good boy-scout, I was thinking how to clean it up a little bit.

My suggestion is to replace the if/else-if/else with a match like this:

```parentTupleSpaces match {
  case Seq() => ...no parent case
  case Seq((_, parentStateHash)) => ...single parent case
  case Seq((initParent, initStateHash), _) => ...
}```

...which is more readable and probably clean enough for now.